src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -270,44 +272,57 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
             }
             else if (!string.IsNullOrEmpty(assignmentContext.AzureFilesConnectionString))
             {
-                await MountCifs(assignmentContext.AzureFilesConnectionString, assignmentContext.AzureFilesContentShare, "/home");
+                await _meshInitServiceClient.MountCifs(assignmentContext.AzureFilesConnectionString, assignmentContext.AzureFilesContentShare, "/home");
             }
 
-            // Irrespective of deployment mechanism mount the share for user data files.
-            if (assignmentContext.IsUserDataMountEnabled())
+            // BYOS
+            foreach (var storageInfoEnvironmentVariable in assignmentContext.GetBYOSEnvironmentVariables())
             {
-                await MountUserData(assignmentContext);
+                var azureStorageInfoValue = AzureStorageInfoValue.FromEnvironmentVariable(storageInfoEnvironmentVariable);
+                if (azureStorageInfoValue != null)
+                {
+                    // Best effort. Mount as many shares as we can.
+                    await MountStorageAccount(azureStorageInfoValue);
+                }
+                else
+                {
+                    _logger.LogWarning($"Failed to parse BYOS account information {storageInfoEnvironmentVariable.Key}");
+                }
```

Comment:
You should be able to mount those in parallel 

```csharp
await Task.WhenAll(assignmentContext.GetBYOSEnvironmentVariables()
    .Select(AzureStorageInfoValue.FromEnvironmentVariable)
    .Where(i => i != null)
    .Select(MountStorageAccount))
```

or if you wanna log the failure you can just add the tasks to a List and Task.WhenAll at the end. 


Failure mode will be different though if the payload is invalid (like if they try to mount different volumes to the same path. Not sure if the API validates that.)

Reply:
I though of making this parallel, but wasn't sure if the init server would have any issues with that. If that is not a concern will switch to Task.WhenAll.

Yes. the api has 2 validations (block duplicate mount paths (existing) and block any mount path under /home (being added now))

Reply:
It should be fine from the init server prospective.